pkg/server/commands/reverseexpand/reverse_expand_weighted.go

```diff
@@ -677,48 +677,48 @@ func (c *ReverseExpandQuery) intersectionHandler(
 	sourceUserType string,
 	resolutionMetadata *ResolutionMetadata,
 ) error {
-	intersectionEdgeComparison, err := typesystem.GetEdgesForIntersection(edges, sourceUserType)
+	intersectionEdges, err := typesystem.GetEdgesForIntersection(edges, sourceUserType)
 	if err != nil {
 		return fmt.Errorf("%w: operation: intersection: %s", ErrLowestWeightFail, err.Error())
 	}
 
-	if !intersectionEdgeComparison.DirectEdgesAreLeastWeight && intersectionEdgeComparison.LowestEdge == nil {
-		// no need to go further because list objects must return empty
+	// if no edges to call LO, return nill
+	if len(intersectionEdges.LowestEdges) == 0 {
 		return nil
 	}
 
-	lowestWeightEdges := []*weightedGraph.WeightedAuthorizationModelEdge{intersectionEdgeComparison.LowestEdge}
-
-	if intersectionEdgeComparison.DirectEdgesAreLeastWeight {
-		lowestWeightEdges = intersectionEdgeComparison.DirectEdges
-	}
-
 	tmpResultChan := make(chan *ReverseExpandResult, listObjectsResultChannelLength)
-
-	siblings := intersectionEdgeComparison.Siblings
-	usersets := make([]*openfgav1.Userset, 0, len(siblings)+1)
-
-	if !intersectionEdgeComparison.DirectEdgesAreLeastWeight && len(intersectionEdgeComparison.DirectEdges) > 0 {
-		// direct weight is not the lowest edge. Therefore, need to call check against directly assigned types.
-		usersets = append(usersets, typesystem.This())
-	}
-
-	for _, sibling := range siblings {
-		userset, err := c.typesystem.ConstructUserset(sibling)
+	intersectEdges := intersectionEdges.SiblingEdges
+	usersets := make([]*openfgav1.Userset, 0, len(intersectEdges))
+	for _, edges := range intersectEdges {
+		// no matter how many direct edges, for typesystem only required this
```

Comment:
Can we add a bit more context on the comment?

Also why have this case if its already handled by `typesystem.ConstructUserset`? I don't think we gain much there or I'm missing something.